rpc/http.go

```diff
@@ -313,3 +96,55 @@ func (client *httpClient) Close() {
 func (client *httpClient) SupportedModules() (map[string]string, error) {
 	return SupportedModules(client)
 }
+
+// httpReadWriteNopCloser wraps a io.Reader and io.Writer with a NOP Close method.
+type httpReadWriteNopCloser struct {
+	io.Reader
+	io.Writer
+}
+
+// Close does nothing and returns always nil
+func (t *httpReadWriteNopCloser) Close() error {
+	return nil
+}
+
+// newJSONHTTPHandler creates a HTTP handler that will parse incoming JSON requests,
+// send the request to the given API provider and sends the response back to the caller.
+func newJSONHTTPHandler(srv *Server) http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		if r.ContentLength > maxHTTPRequestContentLength {
+			http.Error(w,
+				fmt.Sprintf("content length too large (%d>%d)", r.ContentLength, maxHTTPRequestContentLength),
+				http.StatusRequestEntityTooLarge)
+			return
+		}
+
+		w.Header().Set("content-type", "application/json")
+
+		// create a codec that reads direct from the request body until
+		// EOF and writes the response to w and order the server to process
+		// a single request.
+		codec := NewJSONCodec(&httpReadWriteNopCloser{r.Body, w})
+		srv.ServeSingleRequest(codec)
+		codec.Close()
```

Comment:
Minor nitpick, I think it might be better here to do it like:

```
		codec := NewJSONCodec(&httpReadWriteNopCloser{r.Body, w})
		defer codec.Close()
		srv.ServeSingleRequest(codec)
```

to ensure everything's cleaned up even in the cse of a panic or something like that.